newIDE/app/src/ExpressionAutocompletion/index.js

```diff
@@ -207,17 +220,77 @@ const getAutocompletionsForObject = function(
     ...filteredObjectsList.map(({ object }) => ({
       kind: 'Object',
       completion: object.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       object,
       addDot,
     })),
     ...filteredGroupsList.map(({ group }) => ({
       kind: 'Object',
       completion: group.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       addDot,
     })),
   ];
 };
 
+const getAutocompletionsForText = function(
+  expressionAutocompletionContext: ExpressionAutocompletionContext,
+  completionDescription: gdExpressionCompletionDescription
+): Array<ExpressionAutocompletion> {
+  const prefix: string = completionDescription.getPrefix();
+  const type: string = completionDescription.getType();
+  //const objectName: string = completionDescription.getObjectName();
+  const {
+    globalObjectsContainer,
+    objectsContainer,
+  } = expressionAutocompletionContext;
+
+  let autocompletionTexts: string[] = [];
+  if (type === 'layer') {
+    const layout = (objectsContainer: gdLayout);
+    for (let index = 0; index < layout.getLayersCount(); index++) {
+      autocompletionTexts.push(`"${layout.getLayerAt(index).getName()}"`);
+    }
+  } else if (type === 'sceneName') {
+    const project = (globalObjectsContainer: gdProject);
+    for (let index = 0; index < project.getLayoutsCount(); index++) {
+      autocompletionTexts.push(`"${project.getLayoutAt(index).getName()}"`);
+    }
+  } else if (type === 'stringWithSelector') {
+    //TODO handle stringWithSelector autocompletion
+  } else if (type === 'joyaxis') {
+    //TODO handle joyaxis autocompletion
+  }
+  //TODO add missing string types in Core\GDCore\Extensions\Metadata\ParameterMetadata.h
+
+  const filterStringList = (
+    list: Array<string>,
+    searchText: string
+  ): ObjectWithContextList => {
```

Comment:
`Array<string>`

Reply:
Also you might as well move this function outside of the function, to avoid re-declaring (so re-allocating) it at each call.